### DISCUSSION TITLE
Option to use RTD (PT100 or PT1000)

### DISCUSSION THREAD

**Discussion by TheMangrove** [Zer0-bit/gaggiuino#547] [Q&A] [answered]
Is there any plan to add an option to use an RTD sensor for better accuracy, and/or was there a specific reason that a K-type thermocouple was chosen over using an RTD other than cost/availability? 

From what I can tell the K-type should have a faster response time, however an RTD is much more accurate. I was planning on modifying my machine with an ESP32 as a PID controller using an RTD1000 however having stumbled on this project it makes way more sense to implement this project, however I'm curious if using an RTD would improve things, or if it's just better on paper but wouldn't really offer any noticeable improvement.

**Comment by banoz** [comment, marked as answer]
The mod doesn't require accuracy beyond what's provided by a thermocouple. And RTD is much more expensive.

**Comment by TheMangrove** [comment]
So from what I can find, a K-Type thermocouple in the 0-200C range has an accuracy of +/-2.2 C. When this project started, RTD's were pretty expensive, however thanks to the abundance of 3D printers, you can now find them for ~$15. You'd need to add another $3 for a 31865 board. 

You could go cheaper if you went with a PT100 instead of a 3 wire PT1000 which I would agree is likely overkill. 

So you'd be looking at an extra $15 or less to go from 2 degree accuracy to less than a degree accuracy.

I should also add that K types have a greater tendency to drift over time compared to RTDs, so long term that accuracy gap will increase.
